pkg/format/format.go

```diff
@@ -133,52 +203,90 @@ func format(d, label, fsType string) error {
 	return nil
 }
 
+func isBlockDevice(d *os.FileInfo) bool {
+	// this probably shouldn't be so hard
+	// but d.Mode()&os.ModeDevice == 0 doesn't work as expected
+	mode := (*d).Sys().(*syscall.Stat_t).Mode
+	return (mode & syscall.S_IFMT) == syscall.S_IFBLK
+}
+
 // return a list of all available drives
 func findDrives() {
 	drives = make(map[string]bool)
 	driveKeys = []string{}
 	ignoreExp := regexp.MustCompile(`^loop.*$|^nbd.*$|^[a-z]+[0-9]+$`)
 	devs, _ := ioutil.ReadDir("/dev")
 	for _, d := range devs {
-		// this probably shouldn't be so hard
-		// but d.Mode()&os.ModeDevice == 0 doesn't work as expected
-		mode := d.Sys().(*syscall.Stat_t).Mode
-		if (mode & syscall.S_IFMT) != syscall.S_IFBLK {
+		if isBlockDevice(&d) {
+			if verboseVar {
+				log.Printf("/dev/%s is a block device", d.Name())
+			}
+		} else {
+			if verboseVar {
+				log.Printf("/dev/%s is not a block device", d.Name())
+			}
 			continue
 		}
 		// ignore if it matches regexp
 		if ignoreExp.MatchString(d.Name()) {
+			if verboseVar {
+				log.Printf("ignored device /dev/%s during drive autodetection", d.Name())
+			}
 			continue
 		}
 		driveKeys = append(driveKeys, filepath.Join("/dev", d.Name()))
 	}
-	sort.Strings(driveKeys)
-	for _, d := range driveKeys {
-		drives[d] = true
-	}
 }
 
 func init() {
+	flag.BoolVar(&forceVar, "force", false, "Force format of specified single device (default false)")
 	flag.StringVar(&labelVar, "label", "", "Disk label to apply")
 	flag.StringVar(&fsTypeVar, "type", "ext4", "Type of filesystem to create")
+	flag.BoolVar(&verboseVar, "verbose", false, "Enable verbose output (default false)")
+}
+
+func verifyBlockDevice(device string) error {
+	d, err := os.Stat(device)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("%s does not exist", device)
+	}
+	if !isBlockDevice(&d) {
+		return fmt.Errorf("%s is not a block device", device)
+	}
+	// passed checks
+	return nil
 }
 
 func main() {
 	flag.Parse()
 
-	findDrives()
-
 	if flag.NArg() > 1 {
 		log.Fatalf("Too many arguments provided")
 	}
 
 	if flag.NArg() == 0 {
+		// auto-detect drives if a device to format is not explicitly specified
+		findDrives()
```

Comment:
auto-detecting drives only if there are no explicit devices specified to consider singly